### PR TITLE
Use glib's find in path

### LINKED ISSUE #60: Automatically detect the zenity bin path required by VSTGUI X11::FileSelector
Currently there is a `zenity` hard-coded path in both [VSTGUI] and editor's [NativeHelpers].
It would be possible to auto-detect them by [searching in PATH] for systems like NixOS.
See sfztools/sfizz#505.

[VSTGUI]: https://github.com/sfztools/vstgui/blob/f4fe0d0b1e53443342ba732a0e715ce5c96b49df/vstgui/lib/platform/linux/x11fileselector.cpp#L20
[NativeHelpers]: https://github.com/sfztools/sfizz/blob/develop/plugins/editor/src/editor/NativeHelpers.cpp#L115
[searching in PATH]: https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#g-find-program-in-path

### PR DESCRIPTION
**Pull request by paulfd**
Unlike the similar VSTGUI change, we already include gio in sfizz so we can use @redtide 's proposed function.
Closes: sfztools/sfizz-ui#60 (along with the proposed VSTGUI patch on our fork)
I believe nothing calls this code atm, I have to try it out manually.

### CODE REVIEW

**Comment by redtide** [comment]
I forgot some details about this but the code should be also changed in the same way upstream on VSTGUI.

**Comment by paulfd** [comment]
Yes I have an equivalent PR in our fork.

Jul 25, 2021 12:21:46 redtide ***@***.***>:

File: plugins/editor/src/editor/NativeHelpers.cpp [hunk 1]
```diff
@@ -198,12 +198,16 @@ static std::vector<char *> createForkEnviron()
     return newEnv;
 }
 
-static constexpr char zenityPath[] = "/usr/bin/zenity";
+const char* zenityPath()
+{
+    static const char* zenityPath = g_find_program_in_path("zenity");
```
Comment (jpcima) [inline plugins/editor/src/editor/NativeHelpers.cpp]: Release glib's path with `g_free` after done.
other: zenityPath can be null if not found; fallback to the hardcoded value?

Comment (paulfd) [reply plugins/editor/src/editor/NativeHelpers.cpp]: Better like this?

Comment (jpcima) [reply plugins/editor/src/editor/NativeHelpers.cpp]: It's good

**Pull request merged**